test/e2e/specs/onboarding/start__sell.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @group calypso-pr
+ */
+
+import {
+	DataHelper,
+	DomainSearchComponent,
+	EditorPage,
+	GeneralSettingsPage,
+	SignupPickPlanPage,
+	StartSiteFlow,
+	TestAccount,
+} from '@automattic/calypso-e2e';
+import { Browser, Page } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Hero Flow: Sell' ), () => {
+	const themeName = 'Dorna';
+	const storeName = DataHelper.getBlogName();
+	const tagline = `${ storeName } tagline`;
+	const siteSlug = storeName + '.wordpress.com';
+
+	let page: Page;
+	let startSiteFlow: StartSiteFlow;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+		startSiteFlow = new StartSiteFlow( page );
+
+		const testAccount = new TestAccount( 'calypsoPreReleaseUser' );
+		await testAccount.authenticate( page );
+
+		/**
+		 * Create a new site using existing test user 'calypsoPreReleaseUser'
+		 * with a WordPress.com domain and a free plan
+		 *
+		 * @param storeName Site subdomain
+		 */
+		await page.goto( DataHelper.getCalypsoURL( '/start' ) );
+
+		// Select a free domain
+		const domainSearchComponent = new DomainSearchComponent( page );
+		await domainSearchComponent.search( storeName );
+		await domainSearchComponent.selectDomain( '.wordpress.com' );
+
+		// Select the free plan
+		const signupPickPlanPage = new SignupPickPlanPage( page );
+		await signupPickPlanPage.selectPlan( 'Start with Free' );
```

Comment:
You mentioned in the P2 that you'd like a lot more of the operations to be handled via API calls. I totally agree - the `beforeAll` and `afterAll` hooks can quickly become bloated and start doing a lot more than it should.

My question: how much of the calls that you've put into the before and after hooks would you shift to the API calls? A little backstory, I'm currently working on adding the very feature you mention here into the framework in order to implement some e2e coverage for another feature and this has been a topic that's been on my mind recently.

Reply:
Unless the step is part of the actual test, it should be done in the most stable way, usually in the back end. I think this for a few reasons:

- It's faster to run API calls than to load the front-end; if there's a back-end problem, the front end will eventually fail anyway and be slower to do it.
- The back end is less likely to change due to an a/b test and probably won't change as often as the front end.
- I prefer a more atomic test style of getting to what we want to test as quickly as possible. Any additional steps to get to the actual test are just a chance for an unrelated test flake. We aren't trying to test if site creation is working here. There should be a different test failure in a spec specifically for site creation tests.

The argument could be made that a single flow test should validate that creating a new site will drop the user into the hero flow. However, that shouldn't be a beforeAll process; that should be one scenario in the test suite that we do once. We shouldn't need it for all the branching logic tests.